platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClient.java

```diff
@@ -0,0 +1,353 @@
+            transformerProperties.addOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+            write = new DataOutputStream(connection.getOutputStream());
+            write.writeBytes(XMLUtils.format(requestDocument, transformerProperties));
+            write.flush();
+            write.close();
+
+            int statusCode = connection.getResponseCode();
+            if (!reportConnectionStatus(statusCode)) {
+                // If connection is unsuccessful, do not continue and return null.
+                return null;
+            }
+
+        } catch (IOException e) {
+            LOGGER.error("Invalid connection, unable to send request to {}",
+                    externalAttributeStoreUrl);
+            throw new AttributeQueryException("Could not create connection.", e);
+        }
+
+        try {
+            Assertion assertion = null;
+            DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+            documentBuilderFactory.setNamespaceAware(true);
+            Document responseDocument = documentBuilderFactory.newDocumentBuilder()
+                    .parse(connection.getInputStream());
+
+            // Print Response
+            if (LOGGER.isTraceEnabled()) {
+                printXML("SAML Response:\n {}", responseDocument);
+            }
+
+            // Extract Response from Soap message.
+            Node responseNode = responseDocument.getElementsByTagNameNS(SAML2_PROTOCOL, "Response")
+                    .item(0);
+            Element responseElement = (Element) responseNode;
+
+            Unmarshaller unmarshaller = Configuration.getUnmarshallerFactory()
+                    .getUnmarshaller(responseElement);
+
+            Response response = (Response) unmarshaller.unmarshall(responseElement);
+            LOGGER.debug("Successfully marshalled Element to SAML Response.");
+            if (response.getStatus().getStatusCode().getValue().equals(SAML2_SUCCESS)) {
+                LOGGER.debug("Successful response, retrieved attributes.");
+                // Should only have one assertion.
+                assertion = response.getAssertions().get(0);
+
+            } else {
+                reportError(response.getStatus());
+            }
+            return assertion;
+        } catch (IOException | ParserConfigurationException | SAXException e) {
+            throw new AttributeQueryException(
+                    "Unable to parse request string into an XML document.", e);
+        } catch (UnmarshallingException e) {
+            throw new AttributeQueryException("Unable to marshall Element to SAML Response.", e);
+        }
+    }
+
+    /**
+     * If response is unsuccessful, report the cause of the error.
+     *
+     * @param status Status of response object.
+     */
+    private void reportError(Status status) throws AttributeQueryException {
+        if (status.getStatusCode().getValue().equals(SAML2_UNKNOWN_ATTR_PROFILE)) {
```

Comment:
Since Java7, switch with String blocks has been supported. This would read much cleaner as:
```
switch (status.getStatusCode().getValue)) {
 case SAML2_UNKNOWN_ATTR_PROFILE: 
...
}
```